betterconf/caster.py

```diff
@@ -27,7 +29,7 @@ def cast(self, val: str) -> typing.Union[str, VT]:
                     return self.ABLE_TO_CAST[key]
                 elif isinstance(key, str) and val.lower() == key:
                     return self.ABLE_TO_CAST[key]
-            return val
+            return ImpossibleToCastError(val, self)
```

Comment:
Why do you `return` the error instead of `raise`-ing it? This looks kind of weird for me.